### PR TITLE
[Google AdWords Scripts] introduce typings

### PR DESCRIPTION
**Pull request by jafaircl**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the readme.
- [x] Avoid common mistakes.
- [x] Run npm run lint package-name (or tsc if no tslint.json is present).

If adding a new definition:

- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with dts-gen --dt, not by basing it on an existing project.
- [x] tslint.json should be present, and tsconfig.json should have noImplicitAny, noImplicitThis, and strictNullChecks set to true.

Types for writing Google AdWords Scripts using typescript. Google AdWords Scripts are written in JavaScript and run in an environment inside the Google AdWords interface. These types help write them locally using typescript and use the JavaScript output in the interface.

### CODE REVIEW

File: types/google-adwords-scripts/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1564 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="google-apps-script" />
+
+// Generics
+interface AdWordsEntity {
+    getEntityType?(): string;
+}
+
+interface AdWordsIterator<E extends AdWordsEntity | {}> {
```
Comment (RyanCavanaugh) [inline types/google-adwords-scripts/index.d.ts]: `E extends AdWordsEntity | {}` is the same as `E` - what's the intent here?

File: types/google-adwords-scripts/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,1564 @@
...
+    function select(account: AdWordsEntity): void; // TODO: ManagedAccount
+}
+
+declare namespace Analytics { }
+
+declare namespace BigQuery { }
+
+declare namespace Calendar { }
+
+declare namespace FusionTables { }
```
Comment (RyanCavanaugh) [inline types/google-adwords-scripts/index.d.ts]: This is unused, what's it for?

File: types/google-adwords-scripts/index.d.ts (hunk 1 above)
Comment (jafaircl) [inline types/google-adwords-scripts/index.d.ts]: Good point. I'll clean that up.

File: types/google-adwords-scripts/index.d.ts (hunk 2 above)
Comment (jafaircl) [inline types/google-adwords-scripts/index.d.ts]: They are the Advanced APIs available. I wanted to make sure using them didn't throw an error at least until they can be fleshed out with their methods. More info about the Advanced APIs is here: https://developers.google.com/adwords/scripts/docs/features/advanced-apis

**Pull request merged**